src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1447,27 +1461,95 @@ protected override void AfterLeftChildHasBeenVisited(BoundBinaryOperator binary)
 
                         if (operandComparedToNull.Type?.IsReferenceType == true)
                         {
-                            int slot = MakeSlot(operandComparedToNull);
-
-                            if (slot > 0)
+                            Normalize(ref this.State);
+                            var slotBuilder = ArrayBuilder<int>.GetInstance();
+                            getOperandSlots(operandComparedToNull, slotBuilder);
+                            if (slotBuilder.Count != 0)
                             {
-                                if (slot >= this.State.Capacity) Normalize(ref this.State);
-
                                 Split();
-
-                                if (op == BinaryOperatorKind.Equal)
-                                {
-                                    this.StateWhenFalse[slot] = true;
-                                }
-                                else
+                                ref LocalState state = ref (op == BinaryOperatorKind.Equal) ? ref this.StateWhenFalse : ref this.StateWhenTrue;
+                                foreach (int slot in slotBuilder)
                                 {
-                                    this.StateWhenTrue[slot] = true;
+                                    state[slot] = true;
                                 }
                             }
                         }
                     }
                 }
             }
+
+            void getOperandSlots(BoundExpression operand, ArrayBuilder<int> slotListBuilder)
+            {
+                Debug.Assert(operand != null);
+                Debug.Assert(_lastConditionalAccessSlot == -1);
+                int slot;
```

Comment:
Consider moving `slot` declaration closer to the use - either inside each `case` statement or at least inside the `do` loop.